### PR TITLE
Support input tracking loss and selection events in XR

### LINKED ISSUE #24192: Support selection events in WebXR
https://immersive-web.github.io/webxr/#dom-xrsession-onselectstart
https://immersive-web.github.io/webxr/#dom-xrsession-onselect
https://immersive-web.github.io/webxr/#dom-xrsession-onselectend

[Events](https://doc.servo.org/webxr_api/enum.Event.html) need some way of signaling selection.

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes https://github.com/servo/servo/issues/24192

Requires https://github.com/servo/webxr/pull/65

r? @jdm @asajeffrey

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/24362)

### CODE REVIEW

File: components/script/dom/xrsession.rs [hunk 1]
```diff
@@ -170,6 +171,50 @@ impl XRSession {
...
             },
+            XREvent::Select(input, kind, frame) => {
+                let source = self
+                    .input_sources
+                    .borrow_mut()
+                    .iter()
+                    .find(|s| s.id() == input)
+                    .map(|x| DomRoot::from_ref(&**x));
+                if let Some(source) = source {
+                    let frame = XRFrame::new(&self.global(), self, frame);
```
Comment (asajeffrey) [inline components/script/dom/xrsession.rs]: Do you have a spec link for this? Generating a new frame here seems odd, since we're not in an rAF.

**Review by asajeffrey — COMMENTED** [review]
Just a minor question. otherwise LGTM. You can r=me. I'm a bit surprised there's no tests that now pass.

**Comment by Manishearth** [comment]
Tests won't pass since we've not added headless support for this yet.

File: components/script/dom/xrsession.rs (hunk 1 above)
Comment (Manishearth) [reply components/script/dom/xrsession.rs]: https://immersive-web.github.io/webxr/#primary-action , each event comes with a frame


There's a concept of a non-animation frame "frame", which is basically for updating input poses.

**Pull request merged**